docs/source/jit_python_values.rst

```diff
@@ -0,0 +1,466 @@
+.. contents::
+    :local:
+    :depth: 2
+
+
+.. testsetup::
+
+    # These are hidden from the docs, but these are necessary for `doctest`
+    # since the `inspect` module doesn't play nicely with the execution
+    # environment for `doctest`
+    import torch
+
+    original_script = torch.jit.script
+    def script_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_script(obj, *args, **kwargs)
+    torch.jit.script = script_wrapper
+
+    original_trace = torch.jit.trace
+    def trace_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_trace(obj, *args, **kwargs)
+    torch.jit.trace = trace_wrapper
+
+.. _python-builtin-functions-values-resolution:
+
+Resolution Rules
+================
+When given a Python value, TorchScript attempts to resolve them in following five different ways:
+
+* Compilable Python Implementation: 
+    * When a Python value is backed by a Python implementation that can be compiled by TorchScript, TorchScript compiles and uses the underlying Python function.
```

Comment:
What if the implementation is not a function? I'm confused by the words `underlying Python function`.

Reply:
Changed to "underlying Python implementation"